src/Microsoft.NetCore.Analyzers/Core/Security/UseSharedAccessProtocolHttpsOnly.cs

```diff
@@ -0,0 +1,140 @@
+
+                compilationStartAnalysisContext.RegisterOperationBlockStartAction(operationBlockStartContext =>
+                {
+                    var owningSymbol = operationBlockStartContext.OwningSymbol;
+                    var interproceduralAnalysisConfig = InterproceduralAnalysisConfiguration.Create(
+                                                            operationBlockStartContext.Options,
+                                                            SupportedDiagnostics,
+                                                            defaultInterproceduralAnalysisKind: InterproceduralAnalysisKind.None,
+                                                            cancellationToken: operationBlockStartContext.CancellationToken,
+                                                            defaultMaxInterproceduralMethodCallChain: 1);
+
+                    operationBlockStartContext.RegisterOperationAction(operationAnalysisContext =>
+                    {
+                        var invocationOperation = (IInvocationOperation)operationAnalysisContext.Operation;
+                        var methodSymbol = invocationOperation.TargetMethod;
+
+                        if (methodSymbol.Name != "GetSharedAccessSignature")
+                        {
+                            return;
+                        }
+
+                        var namespaceSymbol = methodSymbol.ContainingNamespace;
+
+                        if (namespaceSymbol == null)
+                        {
+                            return;
+                        }
+
+                        var namespaceQualifiedName = namespaceSymbol.ToDisplayString();
+
+                        if (namespaceQualifiedName != "Microsoft.WindowsAzure.Storage" &&
+                            !namespaceQualifiedName.StartsWith("Microsoft.WindowsAzure.Storage.", StringComparison.Ordinal))
+                        {
+                            return;
+                        }
+
+                        var valueContentAnalysisResultOpt = ValueContentAnalysis.GetOrComputeResult(
+                                                                                    invocationOperation.GetTopmostParentBlock().GetEnclosingControlFlowGraph(),
+                                                                                    owningSymbol,
+                                                                                    wellKnownTypeProvider,
+                                                                                    interproceduralAnalysisConfig,
+                                                                                    out var copyAnalysisResult,
+                                                                                    out var pointsToAnalysisResult);
+                        var typeSymbol = methodSymbol.ContainingType;
+
+                        if (typeSymbol.Name != "CloudStorageAccount")
```

Comment:
Again, avoid name comparisons in favour of symbol comparisons. You should fetch type symbol using GetTypeByMetadataName upfront.